addons/test_mail/static/tests/systray_activity_menu_tests.js

```diff
@@ -94,24 +88,22 @@ QUnit.test("activity menu widget: activity menu with 2 models", async (assert) =
     await click(".o_menu_systray i[aria-label='Activities']");
     await contains(".o-mail-ActivityMenu");
     await contains(".o-mail-ActivityMenu .o-mail-ActivityGroup", { count: 2 });
-    assert.ok(
-        containsN(document, ".o-mail-ActivityMenu .o-mail-ActivityGroup", ["res.partner", "0 Late"])
-    );
-    assert.ok(
-        containsN(document, ".o-mail-ActivityMenu .o-mail-ActivityGroup", ["res.partner", "1 Today"])
-    );
-    assert.ok(
-        containsN(document, ".o-mail-ActivityMenu .o-mail-ActivityGroup", ["res.partner", "0 Future"])
-    );
-    assert.ok(
-        containsN(document, ".o-mail-ActivityMenu .o-mail-ActivityGroup", ["mail.test.activity", "1 Late"])
-    );
-    assert.ok(
-        containsN(document, ".o-mail-ActivityMenu .o-mail-ActivityGroup", ["mail.test.activity", "1 Today"])
-    );
-    assert.ok(
-        containsN(document, ".o-mail-ActivityMenu .o-mail-ActivityGroup", ["mail.test.activity", "2 Future"])
-    );
+    await contains(".o-mail-ActivityMenu .o-mail-ActivityGroup", {
+        contains: [
+            ["div[name='activityTitle']", { text: "res.partner" }],
+            ["span", { text: "0 Late" }],
+            ["span", { text: "1 Today" }],
```

Comment:
I'm curious, does this specifically check that we have "1 Today" **for res.partner** ?

As it seems to independently check each selector, so you could have "1 Today" for "mail.test.activity" and this check would still be green, right?

I was probably already the case before though, maybe not needed to change everything.

Reply:
Hello, just answering as I'm passing-by, all the sub-contains need to be all children of the same `.o-mail-ActivityMenu .o-mail-ActivityGroup` at the same time.
So it specifically checks for one notification item that has res.partner, 0 late and 1 today at the same time.

The custom implementation of `containsN` that was in this test before the diff made the same kind of check actually.

Reply:
Hello @seb-odoo 

Ok great then, thanks for the explanations!